src/SmartComponents/BaselinesPage/EditBaseline/DeleteFactModal/DeleteFactModal.js

```diff
@@ -0,0 +1,174 @@
+        let apiBody = editBaselineHelpers.makeDeleteFactsPatch(editBaselineTableData, baselineData);
+
+        let results = await patchBaseline(baselineData.id, apiBody);
+        if (results) {
+            fetchBaselineData(baselineData.id);
+        }
+
+        this.props.toggleModal();
+    }
+
+    buildMessage = (categoryMessage, factMessage) => {
+        let numFactsMessage;
+        let deleteCatMessage;
+        let message;
+
+        if (categoryMessage && factMessage) {
+            numFactsMessage = 'You have selected ' + categoryMessage + ' and ' + factMessage + ' to be deleted.';
+            deleteCatMessage = 'Deleting a category will delete all facts within the category.';
+            message = <div>
+                <span className='display-block'>{ numFactsMessage }</span>
+                <span className='display-block'>{ deleteCatMessage }</span>
+            </div>;
+        } else if (categoryMessage) {
+            numFactsMessage = 'You have selected ' + categoryMessage + ' to be deleted.';
+            deleteCatMessage = 'Deleting a category will delete all facts within the category.';
+            message = <div>
+                <span className='display-block'>{ numFactsMessage }</span>
+                <span className='display-block'>{ deleteCatMessage }</span>
+            </div>;
+        } else if (factMessage) {
+            numFactsMessage = 'You have selected ' + factMessage + ' to be deleted.';
+            message = <div>
+                <span className='display-block'>{ numFactsMessage }</span>
+            </div>;
+        }
+
+        return message;
+    }
+
+    countFacts = () => {
+        const { editBaselineTableData } = this.props;
+        let categories = 0;
+        let facts = 0;
+        let categoryMessage;
+        let factMessage;
+        let message;
+
+        editBaselineTableData.forEach(function(fact) {
+            if (fact.selected) {
+                if (Array.isArray(fact[2])) {
+                    categories += 1;
+                } else {
+                    facts += 1;
+                }
+            } else if (Array.isArray(fact[2])) {
+                fact[2].forEach(function(subFact) {
+                    if (subFact.selected) {
+                        facts += 1;
+                    }
+                });
+            }
+        });
+
+        if (categories === 1) {
+            categoryMessage = '1 category';
+        } else if (categories > 1) {
+            categoryMessage = categories + ' categories';
+        }
+
+        if (facts === 1) {
+            factMessage = '1 fact';
+        } else if (facts > 1) {
+            factMessage = facts + ' facts';
+        }
+
+        message = this.buildMessage(categoryMessage, factMessage);
+
+        return message;
+    }
+
+    render() {
+        const { modalOpened } = this.state;
+        const { deleteFact, isCategory } = this.props;
+        let message;
+
+        if (deleteFact) {
```

Comment:
What is the deleteFact property? Is this the use case to handle the kabob issuing a delete?

Reply:
This is a function passed from the FactKebab file that helps keeps the modal as dumb as possible about the contents of the fact. https://github.com/RedHatInsights/drift-frontend/blob/4237ff92a1346ba9683e58aebefeda287e4667a3/src/SmartComponents/BaselinesPage/EditBaseline/FactKebab/FactKebab.js#L119